app/connectors/CdsFileUploadConnector.scala

```diff
@@ -43,4 +43,8 @@ class CdsFileUploadConnector @Inject()(appConfig: AppConfig, httpClient: HttpCli
         }
         notificationOpt
       }
+
+  def getDeclarationStatus(mrn: MRN)(implicit hc: HeaderCarrier): Future[HttpResponse] =
+    httpClient.GET[HttpResponse](cdsFileUploadConfig.fetchDeclarationStatusEndpoint(mrn.value))
```

Comment:
Out of curiosity... how are potential exceptions thrown by HttpClient handled here?

Reply:
Yes